File: JavaScript solutions/442. Find All Duplicates in an Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function (nums) {
  const ans = [];

  for (let i = 0; i < nums.length; i++) {
    const ind = Math.abs(nums[i]) - 1;
    if (nums[ind] > 0) {
      nums[ind] = -nums[ind];
    } else {
      ans.push(Math.abs(nums[i]));
    }
  }

  return ans;
};

File: C++ solutions/442. Find All Duplicates in an Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int>ans;
        for(int i=0;i<nums.size();i++){
            int ind=abs(nums[i])-1;
            if(nums[ind]>0){
                nums[ind]=(-nums[ind]);
            }
            else ans.push_back(abs(nums[i]));
        }
        return ans;
    }
};